packages/sync/src/Queue.php

```diff
@@ -461,7 +696,7 @@ private function validate_checkout( $buffer ) {
 			return new \WP_Error( 'buffer_not_checked_out', 'There are no checked out buffers' );
 		}
 
-		if ( $checkout_id != $buffer->id ) {
+		if ( intval( $checkout_id ) !== intval( $buffer->id ) ) {
```

Comment:
Do we have a test to verify this change won't break anything?

Reply:
yeah this brings me flashbacks 😄

Reply:
reverting until I have time to figure out what this does and how to test properly